src/collisions_updater.cpp

```diff
@@ -0,0 +1,228 @@
+                dc.link1_ = pair_it->first.first;
+                dc.link2_ = pair_it->first.second;
+                dc.reason_ = moveit_setup_assistant::disabledReasonToString( pair_it->second.reason );
+                disabled_collisions.insert(SortableDisabledCollision(dc));
+            }
+        }
+
+        config_data.srdf_->disabled_collisions_.assign(disabled_collisions.begin(), disabled_collisions.end());
+
+        // Update collision_matrix for robot pose's use
+        config_data.loadAllowedCollisionMatrix();
+
+        config_data.srdf_->writeSRDF(output_path.empty() ? config_data.srdf_path_ : output_path );
+
+    }
+
+};
+
+int main(int argc, char * argv[]){
+
+    std::string config_pkg_path;
+    std::string urdf_path;
+    std::string srdf_path;
+
+    std::string output_path;
+
+    bool include_default = false, include_always = false, keep_old = false, verbose = false;
+
+    double min_collision_fraction = 1.0;
+
+    uint32_t never_trials = 0;
+
+    po::options_description desc("Allowed options");
+    desc.add_options()
+        ("help", "show help")
+        ("config-pkg", po::value(&config_pkg_path), "path to moveit config package")
+        ("urdf", po::value(&urdf_path), "path to URDF ( or xacro)")
+        ("srdf", po::value(&srdf_path), "path to SRDF ( or xacro)")
+        ("output", po::value(&output_path), "output path for SRDF")
+
+        ("xacro-args", po::value<std::vector<std::string> >()->composing(), "additional arguments for xacro")
+
+        ("default", po::bool_switch(&include_default),  "disable default colliding pairs")
+        ("always", po::bool_switch(&include_always),  "disable always colliding pairs")
+
+        ("keep", po::bool_switch(&keep_old),  "keep disabled link from SRDF")
+        ("verbose", po::bool_switch(&verbose),  "verbose output")
+
+        ("trials", po::value(&never_trials),  "number of trials for searching never colliding pairs")
+        ("min-collision-fraction", po::value(&min_collision_fraction),  "fraction of small sample size to determine links that are alwas colliding")
+      ;
+
+    po::positional_options_description pos_desc;
+    pos_desc.add("xacro-args", -1);
+
+    po::variables_map vm;
+    po::store(po::command_line_parser(argc, argv).options(desc).positional(pos_desc).run(), vm);
+    po::notify(vm);
+
+    if (vm.count("help")) {
+        std::cout << desc << std::endl;
+        return 1;
+    }
+
+    CollisionUpdater updater;
+
+    if(!config_pkg_path.empty()){
+        if(!updater.loadSetupAssistantConfig(config_pkg_path)){
+            std::cerr << "Could not load config at '" << config_pkg_path << "'" << std::endl;
+            return 1;
+        }
+    }else if (urdf_path.empty() || srdf_path.empty()){
+            std::cerr << "Please provide config package or URDF and SRDF path" << std::endl;
+            return 1;
+    }
+
+    updater.setURDF(urdf_path);
```

Comment:
this would be perfect to note in the comments for future developers to understand